### PR TITLE
updating email Validator

### PR DESCRIPTION
**Pull request by surbhi400**
### 🛠️ Fixes Issue (Number)

Closes #issue_number 
#930 

### ✅ Check List (Check all the applicable boxes) 

- [x] You have linked this PR to the correct issue.
- [x] My code doesn't break any part of the project (Zero Octave-Javascript-Projects).
- [x] This PR does not contain plagiarized content.
- [x] My Addition/Changes works properly and matches the overall repo pattern.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: Public/EmailValidator.html [hunk 1]
```diff
@@ -1,17 +1,41 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
     <meta charset="UTF-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <link rel="stylesheet" href="../assets/css/EmailValidator.css">
+
+    <title>Email validator</title>
```
Comment (Anmol-Baranwal) [inline Public/EmailValidator.html:10]: This should be removed as title tag is included two times.

File: Public/EmailValidator.html [hunk 2]
```diff
@@ -1,17 +1,41 @@
...
+        <div>
+            <button class="btn" id="submit"> Confirm Email </button>
+        </div>
+    </div>
+    <!-- <form action="#">
+        <label></label>
+       <input type="text" name="name" id="name"><br>
         <button id="submit"> Submit </button>
-    </form>
+    </form> -->
```
Comment (Anmol-Baranwal) [inline Public/EmailValidator.html:33]: There is no need for comments that are not related to the email validator. Please remove them if they do not serve any purpose.

File: Public/EmailValidator.html [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
 </body>
 <script src="../assets/Js/EmailValidator.js"></script>
+
+
 </html>
+
+</html>
```
Comment (Anmol-Baranwal) [inline Public/EmailValidator.html:40]: The HTML tag is included twice, so please remove one of them.

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
It does work as intended and can be used as a basic email validator.

However, it has some potential issues that you might want to consider:

1. The function name 'submit' could conflict with the built-in method in HTML form elements with the same name. It is recommended to use a different name to avoid confusion.

2. The code only checks for basic syntax errors and does not verify the existence or validity of the email address itself. It is possible for an email address to pass this check but still be invalid or non-existent.

3. The code does not provide any feedback to the user about the specific issue with the email address. It would be helpful to have more detailed error messages to help users correct their email addresses.

> Please make mentioned changes and also change the function name.

**Comment by Anmol-Baranwal** [comment]
@surbhi400 
I am here to help you. Please feel free to ask any questions, and I will gladly address them.

**Comment by surbhi400** [comment]
Ok i am on it.

**Comment by Anmol-Baranwal** [comment]
@surbhi400, please feel free to get in touch if there are any issues or if you need any assistance. I was just checking in because it has been 2 weeks.

**Comment by surbhi400** [comment]
> @surbhi400, please feel free to get in touch if there are any issues or if you need any assistance. I was just checking in because it has been 2 weeks.

Hey @Anmol-Baranwal, I am really sorry actually I had my internal exams going on and then I broke my laptop screen that is why this took this much time. I sincerely apologize for causing you inconvenience. I will get this done by Monday.

**Comment by Anmol-Baranwal** [comment]
No worries, feel free to complete it whenever you get time. No need to rush.

**Comment by surbhi400** [comment]
Screenshot 2023-04-29 113650
@Anmol-Baranwal could you tell me what could be the problem that it is showing this error. I have already done git pull and it shows its up to date.

**Comment by Anmol-Baranwal** [comment]
> 

It seems like you are facing an error while trying to push changes to a remote branch on GitHub. The error message is suggesting that the remote branch you are trying to push to has some changes that are not present in your local branch.

To resolve this issue, you need to first integrate the remote changes into your local branch by running the following command:

```git
git pull origin emailvalidator
```

This command will fetch the changes from the remote branch to your local branch and merge them.

After resolving any conflicts that may arise during the merge, you can try pushing the changes again using the command:

```git
git push origin emailvalidator
```

This should push your changes to the remote branch successfully. 
> Do let me know if this doesn't works.

File: Public/EmailValidator.html (hunk 3 above)
Comment (surbhi400) [reply Public/EmailValidator.html:28]: I removed the html tag

File: Public/EmailValidator.html (hunk 2 above)
Comment (surbhi400) [reply Public/EmailValidator.html]: I deleted unwanted comments.

**Comment by surbhi400** [comment]
@Anmol-Baranwal It worked. I worked on the changes 1 and 3 you asked me to like now it provides the issue with the email. But I am not very sure of the 2nd change you asked for. Like it still statisfies the one example you mentioned . Can you please guide me that how should I fix that part.

emailValidator(1)
emailValidator(2)
emailValidator(3)

**Comment by Anmol-Baranwal** [comment]
> @Anmol-Baranwal It worked. I worked on the changes 1 and 3 you asked me to like now it provides the issue with the email. But I am not very sure of the 2nd change you asked for. Like it still statisfies the one example you mentioned . Can you please guide me that how should I fix that part.
> 
> emailValidator(1)
> emailValidator(2)
> emailValidator(3)
> 

I will review the changes. If the example works, then these will be enough for me to merge the PR.

**Comment by surbhi400** [comment]
> > @Anmol-Baranwal It worked. I worked on the changes 1 and 3 you asked me to like now it provides the issue with the email. But I am not very sure of the 2nd change you asked for. Like it still statisfies the one example you mentioned . Can you please guide me that how should I fix that part.
> > emailValidator(1)
> > emailValidator(2)
> > emailValidator(3)
> 
> I will review the changes. If the example works, then these will be enough for me to merge the PR.

No sorry, my mistake. I meant to say it does not satisfy the one example you mentioned. And that is why I want you guide me about how to fix it.

**Review by Anmol-Baranwal — APPROVED** [review]
First of all, `example@com.abc` is not a valid email which I don't really know how I missed, but leaving that aside. I would recommend not working on the 2nd point because if you change the logic, you would have to write detailed logic on the user message.

This JavaScript code is a basic email validator that checks the position of "@" and "." in an email input. This code can validate basic email formats and check if "@" and "." are placed appropriately. However, it has some limitations as it does not perform validation for other required elements in an email format, such as username, domain, or top-level domain.

In contrast, more advanced email formats that include characters like + or - in the username, or emails that use newer top-level domain extensions like .club or .store might pass through this basic validation even though they may not be valid.

Therefore, the code can work for simple email formats but is not a comprehensive solution. In practice, the new RegEx solution is a more effective method for email validation that addresses the limitations of the basic validation provided.

I have written a JavaScript validator that passes all necessary checks. You can use it as a reference if you plan on developing an advanced email validator.

```javascript
const emailInput = document.getElementById("email");
const button = document.getElementById("submit");

function ValidateEmail(inputText) {
  const email = inputText.value.trim();

  if (!email) {
    alert("Email cannot be empty");
    return false;
  }

  // you can also use: const regex = /^([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

  if (!regex.test(email)) {
    alert("Invalid email format");
    return false;
  }

  alert("Your Email Id is correct");
  return true;
}
```

I approve the changes, and thanks for sticking through to the end. I appreciate your contribution.

@Astrodevil, do you want to add something? Or should I merge the PR?

**Comment by Astrodevil** [comment]
You can merge it @Anmol-Baranwal You have covered all the points and changes above. Looks good to me for basic functionality of the validator.
